treeherder/perf/models.py

```diff
@@ -611,6 +611,27 @@ class BackfillRecord(models.Model):
     def id(self):
         return self.alert
 
+    @property
+    def repository(self):
+        return self.alert.summary.repository
+
+    def get_context_border_info(self, context_property: str) -> Tuple[str, str]:
+        """
+        Provides border(first and last) information from context based on the property
+        """
+        context = self.get_context()
+        from_info = context[0][context_property]
+        to_info = context[-1][context_property]
+
+        return from_info, to_info
+
+    def get_pushes_in_context_range(self):
```

Comment:
Please provide appropriate type hint for function' s output. It'll help navigate through the code.